### PR TITLE
Handle the "failed to initiate checkout" case and add request specs

### PR DESCRIPTION
**Pull request by anaulin**
(merging into WIP https://github.com/zinc-collective/convene/pull/1031/)

There are a few things that I don't exactly love about this (the weird validation issue, needing the stripe url attributes on checkout), but I think it at least moves things in the right direction.

### CODE REVIEW

File: app/furniture/marketplace/carts/_footer.html.erb [hunk 1]
```diff
@@ -7,7 +7,7 @@
 <tfoot id="cart-footer-<%= cart.id%>" class="bg-gray-50">
   <tr>
     <td></td>
-    <th> <%= link_to("checkout", new_space_room_marketplace_checkout_path(space, room, marketplace), data: { turbo: false }) %> </th>
+    <th> <%= button_to("checkout", [space, room, marketplace, :checkouts], data: { turbo: false }) %> </th>
```
Comment (zspencer) [inline app/furniture/marketplace/carts/_footer.html.erb:10]: Theoretically, `marketplace.location(:checkouts)` should return `[space, room, marketplace, :checkouts]`

**Review by zspencer — APPROVED** [review]
omg, thank you for pulling this out of the controller and into a model! 

Also, I really appreciate that this is now a post and not a GET!

File: app/furniture/marketplace/carts/_footer.html.erb (hunk 1 above)
Comment (anaulin) [reply app/furniture/marketplace/carts/_footer.html.erb:10]: Noted. We can update it on Wednesday.

Comment (anaulin) [reply app/furniture/marketplace/carts/_footer.html.erb:10]: That doesn't seem to work. `marketplace.location(:checkouts)` is creating the path `/spaces/system-test/rooms/entrance-hall/marketplaces/000c9416-a6ff-49f0-9a55-3a86aad16c3a`, which is not right.

Comment (zspencer) [reply app/furniture/marketplace/carts/_footer.html.erb:10]: Blah, I must not have finished applying`HasLocation` to the `Marketplace` namespace.

**Pull request merged**